flink-connectors/flink-connector-elasticsearch-base/src/main/java/org/apache/flink/connector/elasticsearch/table/IndexGeneratorFactory.java

```diff
@@ -176,6 +207,8 @@ private static DynamicFormatter createFormatFunction(
         private static final Pattern dynamicIndexPattern = Pattern.compile("\\{[^\\{\\}]+\\}?");
         private static final Pattern dynamicIndexTimeExtractPattern =
                 Pattern.compile(".*\\{.+\\|.*\\}.*");
+        private static final Pattern dynamicIndexProcTimeExtractPattern =
+                Pattern.compile(".*\\{now\\(\\)\\|.*\\}.*");
```

Comment:
I think following function should also works, right?
`now()` 
`NOW()` 
 `current_timestamp`
 `CURRENT_TIMESTAMP`